### PR TITLE
Fixes singularity not instantly consuming things the center passes over

### LINKED ISSUE #9946: Singularity center object interacts oddly with other objects
If the singularity hits a dense object, it doesn't eat it and continue moving like you would expect, it stops at that object until it consumes it(at which point it starts drifting again, as normal)
The singularity also doesn't consume objects it directly Crossed() as in, if the center object drifts over an object, the object won't be consumed if the singularity drifts fast enough away.

### PR DESCRIPTION
**Pull request by silicons**
fixes #9946

### CODE REVIEW

File: code/modules/power/singularity/singularity.dm [hunk 1]
```diff
@@ -106,6 +106,9 @@
 	consume(A)
 	return
 
+/obj/singularity/Crossed(atom/A)
+	consume(A)
+	return
```
Comment (Cyberboss) [inline code/modules/power/singularity/singularity.dm]: No trailing return

File: code/modules/power/singularity/singularity.dm [hunk 2]
```diff
@@ -106,6 +106,9 @@
 	consume(A)
 	return
 
+/obj/singularity/Crossed(atom/A)
+	consume(A)
```
Comment (Cyberboss) [inline code/modules/power/singularity/singularity.dm]: ..() first

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by ChangelingRain** [comment]
the fact that the singularity doesn't keep drifting after it bumps into something is the major part of the issue there so like, can you make it keep drifting in the direction it was drifting after it bumps something

**Comment by kevinz000** [comment]
@ChangelingRain done it now always returns false on process_spacemove

**Comment by ChangelingRain** [comment]
It already did that, so no, you haven't fixed it.

**Pull request closed without merging**